qlib/Mapper.qm

```diff
@@ -980,8 +1014,8 @@ Mapper mapv(DataMap);
             if (m.code) {
                 try {
                     v = do_list
-                        ? (map m.code($1, getrec(rec, $#)), v)
-                        : m.code(v, rec);
+                        ? (map m.code($1, getrec(rec + crec, $#)), v)
```

Comment:
I want clear view what type of values are at which point where, instead of being prepared to everything. If we want do to it externally, we should enrich 'rec' in proper type (single value with !do_list) and a list with do_list=True